packages/core/src/components/Dropdown/__tests__/driver.js

```diff
@@ -24,6 +24,18 @@ export default class DropdownDriver {
   render() {
     cleanup();
     this.renderResult = render(<Dropdown {...this.props} />);
+    return this.renderResult;
+  }
+
+  async renderWithEffects() {
+    cleanup();
+    await act(async () => {
+      this.renderResult = render(<Dropdown {...this.props} />);
+
+      // If there's asynchronous logic in useEffect, we can await it here
+      await new Promise(resolve => setTimeout(resolve, 100)); // Adjust timeout based on the async logic
```

Comment:
```suggestion
      await new Promise(resolve => setTimeout(resolve, 0)); // Adjust timeout based on the async logic
```
wouldn't it still get the job done? or even RaF?